Pinia

helenahalldiniths opened the pull request:
This will add pinia to the project and a store for handling quizzes

FelixJacobsen commented:
Changes made with kevin

1. Added .value on a few places where a ref is used
2. Switched from 'onBeforeMounted' to 'onMounted'
3. Switched current question from being an object to null
4. Removed all data from questions since it's now being fetched.

LordRekishi commented on README.md:
````diff
@@ -2,21 +2,27 @@
+
 ```sh
 npm install
+npm install pinia
````
No needed, enough with npm install

LordRekishi commented on src/views/HomeView.vue:
```diff
@@ -11,13 +11,14 @@
 import AnswerItems from "../components/AnswerItems.vue";
-import { ref, onBeforeMount } from "vue";
+import { ref, onMounted } from "vue";
+import { useQuizStore } from "@/stores/quiz";
 const answers = ref([]);
```
```suggestion
import { useQuizStore } from "@/stores/quiz";

const answers = ref([]);
```

LordRekishi commented on src/stores/quiz.js:
```diff
@@ -0,0 +1,34 @@
+  const questions = ref([]);
+  const idCurrentQuestion = ref(0);
+
+  //Reaktiv
```
```suggestion
```

LordRekishi commented on src/stores/quiz.js:
```diff
@@ -0,0 +1,34 @@
+  const numberOfQuestions = computed(() => questions.value.length);
+
+  const setQuestions = (newQuestions) => {
+    console.log("Kevins log3", newQuestions);
```
```suggestion
```

LordRekishi commented on src/views/HomeView.vue:
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
-  wordToTranslate.value = "Pear";
-  correctAnswer.value = "päron";
+const setQuestionInfo = () => {
+  console.log("Kevins log", currentQuestion.value);
```
```suggestion
```

LordRekishi commented on src/views/HomeView.vue:
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
+    answeredCorrectly.value = false;
+  } else {
+    // Here we will send the user to the finish/result page.
+    console.log("done");
```
```suggestion
```

LordRekishi commented on src/views/HomeView.vue:
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
+  fetch("questions.json")
+    .then((response) => response.json())
+    .then((data) => {
+      console.log(data);
```
```suggestion
```

LordRekishi commented on src/views/HomeView.vue:
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
+
+      //Set first question
+      currentQuestion.value = quiz.getCurrentQuestion();
+      console.log("Kevins log2", quiz.getCurrentQuestion());
```
```suggestion
```

LordRekishi reviewed (changes requested):
Looks good, let's start with removing some logs and comments. Then I'll look again

helenahalldiniths commented:
Good job Felix! I looked throught the code and I think the logging really helped me understand it! However I agree with Patrik that they should be removed before merging it into the main branch. I removed some logging and added some comments in order to keep track of what we are doing.
